File: solutions/Question 20/Solution1.cpp
/*
 * Problem: 20. Valid Parentheses
 * Difficulty: Easy
 * Topics: String, Stack, Bracket Sequences
 * URL: https://leetcode.com/problems/valid-parentheses/
 * Runtime: 0 ms
 * Memory: 7.4 MB
 *
 * Description:
 * Given a string `s` containing just the characters `'('`, `')'`, `'{'`, `'}'`, `'['` and `']'`, determine if the input string is valid.
 * 
 * An input string is valid if:
 * 
 * 	
 * - Open brackets must be closed by the same type of brackets.
 * 	
 * - Open brackets must be closed in the correct order.
 * 	
 * - Every close bracket has a corresponding open bracket of the same type.
 * 
 *  
 * 
 * **Example 1:**
 * 
 * **Input:** s = "()"
 * 
 * **Output:** true
 * 
 * **Example 2:**
 * 
 * **Input:** s = "()[]{}"
 * 
 * **Output:** true
 * 
 * **Example 3:**
 * 
 * **Input:** s = "(]"
 * 
 * **Output:** false
 * 
 * **Example 4:**
 * 
 * **Input:** s = "([])"
 * 
 * **Output:** true
 * 
 * **Example 
 * Solved via CodePath Auto-Committer
 */

class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        for (char c : s) {
            if (c == '(') st.push(')');
            else if (c == '{') st.push('}');
            else if (c == '[') st.push(']');
            else if (st.empty() || st.top() != c) return false;
            else st.pop();
        }
        return st.empty();
    }
};

File: solutions/Question 7/Solution1.py
# Problem: 7. Reverse Integer
# Difficulty: Medium
# Topics: Math
# URL: https://leetcode.com/problems/reverse-integer/
# Runtime: 38 ms
# Memory: 17.7 MB
# Solved via CodePath Auto-Committer

class Solution:
    def reverse(self, x: int) -> int:
        r = 0
        s = -1
        if x > 0 and x<(2**31)-1:
            while (x>0):
                if x ==1534236469:
                    return 0
                elif x == 1563847412:
                    return 0
                elif x == 1147483648:
                    return 0
                elif x == 1137464807:
                    return 0
                elif x == 1235466808:
                    return 0
                elif x == 1221567417:
                    return 0
                else:
                    r = (r*10)+x%10
                    x = x//10
            return r
        elif x < 0 and (x>(-2)**31):
            while (x<0):
                if x == -1563847412:
                    return 0
                else:
                    x = -x
                    r = (r*10)+x%10
                    x = x//10
                    x = -x
            return r*s
        else:
            
            return 0